# 데이터 핸들링 연습문제 2 (딕셔너리 등)

### 1. 영화진흥위원회 출처의 일별 박스오피스 데이터인 위계가 있는 딕셔너리를 다음과 같이 변수 boxOffice에 담았다. 딕셔너리 boxOffice의 키(key)의 개수와 키들을 출력하시오.

In [1]:
# 출처: 영화진흥위원회 (www.kobis.or.kr)
import requests
URL='https://raw.githubusercontent.com/algoboni/pythoncodebook_visual1/main/data/2_boxOffice.json'
r = requests.get(URL)
boxOffice = r.json()['boxOfficeResult']
print(type(boxOffice))

<class 'dict'>


In [4]:
print(boxOffice.keys(),len(boxOffice.keys()))

dict_keys(['boxofficeType', 'showRange', 'dailyBoxOfficeList']) 3


### 2. 딕셔너리 boxOffice의 키인 'showRange'의 값을 출력하시오.

In [5]:
boxOffice['showRange']

'20231225~20231225'

### 3. 딕셔너리 boxOffice의 키인 'showRange'의 값을 시작날짜와 종료날짜를 요소로 갖는 리스트(ex: ['20231225', '20231225'])로 수정하여 딕셔너리에 적용한 후, 2번 문제와 동일한 방법으로 값을 출력하시오.

In [7]:
boxOffice['showRange'] = [c for c in boxOffice['showRange'].split('~')]
boxOffice['showRange'] 

['20231225', '20231225']

### 4. 딕셔너리 boxOffice의 키인 'dailyBoxOfficeList'의 값을 dboList라는 변수에 담고, 해당 변수의 크기와 데이터 타입을 출력하시오.

In [10]:
dboList = boxOffice['dailyBoxOfficeList']
print(len(dboList),type(dboList))

10 <class 'list'>


### 5. 리스트 dboList의 마지막 요소를 추출하고 dict()를 적용하여 변수 dict11에 담으시오. 또한, 해당 변수를 다음과 같이 편집한 후 출력하시오.
- key 'rnum'의 값을 '11'로 수정
- key 'rank'의 값을 '11'로 수정
- key 'movieNm'의 값을 '알고보는 영화'로 수정
- key 'audiCnt'의 값을 '4300'으로 수정

In [18]:
dict11 = dict(dboList[-1])
dict11['rnum'] = '11'
dict11['rank'] = '11'
dict11['movieNm'] = '알고보는 영화'
dict11['audiCnt'] = '4300'
dict11

{'rnum': '11',
 'rank': '11',
 'rankInten': '1',
 'rankOldAndNew': 'OLD',
 'movieCd': '20235980',
 'movieNm': '알고보는 영화',
 'openDt': '2023-12-13',
 'salesAmt': '47381636',
 'salesShare': '0.3',
 'salesInten': '16715045',
 'salesChange': '54.5',
 'salesAcc': '677175792',
 'audiCnt': '4300',
 'audiInten': '1586',
 'audiChange': '52.2',
 'audiAcc': '70146',
 'scrnCnt': '43',
 'showCnt': '61'}

### 6. 딕셔너리 dict11의 모든 값에 일괄적으로 '**'를 추가한 후, dict11을 출력하시오.

In [19]:
# 모든 값에 '**' 추가
for key in dict11:
    dict11[key] = "**"

# 출력
print(dict11)

{'rnum': '**', 'rank': '**', 'rankInten': '**', 'rankOldAndNew': '**', 'movieCd': '**', 'movieNm': '**', 'openDt': '**', 'salesAmt': '**', 'salesShare': '**', 'salesInten': '**', 'salesChange': '**', 'salesAcc': '**', 'audiCnt': '**', 'audiInten': '**', 'audiChange': '**', 'audiAcc': '**', 'scrnCnt': '**', 'showCnt': '**'}


### 7. 리스트 dboList의 첫번째 요소의 키와 값들을 추출하여 각각 KEYS, VALUES라는 변수에 담아 출력하시오.

In [29]:
KEYS, VALUES = [],[]
for key, value in zip(dboList[0].keys(),dboList[0].values()):
    KEYS.append(key)
    VALUES.append(value)
print(KEYS, VALUES)

['rnum', 'rank', 'rankInten', 'rankOldAndNew', 'movieCd', 'movieNm', 'openDt', 'salesAmt', 'salesShare', 'salesInten', 'salesChange', 'salesAcc', 'audiCnt', 'audiInten', 'audiChange', 'audiAcc', 'scrnCnt', 'showCnt'] ['1', '1', '0', 'OLD', '20203702', '노량: 죽음의 바다', '2023-12-20', '5609307954', '39.9', '-657335829', '-10.5', '22349048606', '549530', '-62118', '-10.2', '2230951', '1845', '6783']


### 8. 앞서 생성한 변수 KEYS와 VALUES의 요소들의 맨 앞에 각각 일괄적으로 'ex) '를 추가한 결과를 각각 KEYS2와 VALUES2에 담고, 이를 키와 값으로 갖는 딕셔너리 dict12를 생성하여 출력하시오.

In [31]:
KEYS2, VALUES2 = [],[]

for key, value in zip(KEYS,VALUES):
    KEYS2.append('ex)'+key)
    VALUES2.append('ex)'+value)

dict12 = dict()
for key, value in zip(KEYS2,VALUES2):
    dict12[key] = value
    
print(dict12)    

{'ex)rnum': 'ex)1', 'ex)rank': 'ex)1', 'ex)rankInten': 'ex)0', 'ex)rankOldAndNew': 'ex)OLD', 'ex)movieCd': 'ex)20203702', 'ex)movieNm': 'ex)노량: 죽음의 바다', 'ex)openDt': 'ex)2023-12-20', 'ex)salesAmt': 'ex)5609307954', 'ex)salesShare': 'ex)39.9', 'ex)salesInten': 'ex)-657335829', 'ex)salesChange': 'ex)-10.5', 'ex)salesAcc': 'ex)22349048606', 'ex)audiCnt': 'ex)549530', 'ex)audiInten': 'ex)-62118', 'ex)audiChange': 'ex)-10.2', 'ex)audiAcc': 'ex)2230951', 'ex)scrnCnt': 'ex)1845', 'ex)showCnt': 'ex)6783'}


### 9. 딕셔너리 boxOffice의 키인 'dailyBoxOfficeList'의 값의 구조를 다음과 같이 하나의 딕셔너리 형태로 변환하여 변수 new_dict에 담고 출력하시오.
{'rnum': ['1', '2', ..., '10'],  
'rank': ['1', '2', ..., '10'], ...  
'showCnt': ['6783', '4657', ... '61'}

In [42]:
to_col = ['rnum','rank','showCnt']
new_dict = dict()
for e in to_col:
    tmp = []
    for ele in boxOffice['dailyBoxOfficeList']:
        tmp.append(ele[e])
    new_dict[e] = tmp
print(new_dict)

{'rnum': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'], 'rank': ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'], 'showCnt': ['6783', '4657', '1796', '2073', '1284', '801', '174', '179', '171', '61']}
